### ISSUE TITLE
ChatSession.onConnectionEstablished() is fired twice after invoking ChatSession.connect()

### ISSUE THREAD

**Issue by marcogrcr** [amazon-connect/amazon-connect-chatjs#124]
**Steps to reproduce:**

```js
import "amazon-connect-chatjs"; // v1.3.1
import {
  ConnectClient,
  StartChatContactCommand,
} from "@aws-sdk/client-connect"; // v3.254.0

// start a chat contact
const client = new ConnectClient({
  region: "us-east-1",
  credentials: {
    accessKeyId: "...",
    secretAccessKey: "...",
    sessionToken: "...",
  },
});

const { ContactId, ParticipantId, ParticipantToken } = await client.send(
  new StartChatContactCommand({
    InstanceId: "...",
    ContactFlowId: "...",
    ParticipantDetails: { DisplayName: "Customer" },
  })
);

// create chat session
const session = connect.ChatSession.create({
  chatDetails: {
    contactId: ContactId,
    participantId: ParticipantId,
    participantToken: ParticipantToken,
  },
  options: { region: "us-east-1" },
  type: connect.ChatSession.SessionTypes.CUSTOMER,
});

// subscribe to `onConnectionEstablished` events
session.onConnectionEstablished((event) => {
  console.log("Connection established:", event);
});

// connect to chat
await session.connect();
```

**Expected result:**

The `onConnectionEstablished` listener gets invoked once:

```
Connection established: {
  chatDetails: { ... },
  data: {}
}
```

**Actual result:**

The `onConnectionEstablished` listener gets invoked twice with different event payloads:

```
Connection established: {
  chatDetails: { ... },
  connectCalled: true,
  connectSuccess: true,
  metadata: null,
  _debug: { webSocketStatus: "Starting" }
}

Connection established: {
  chatDetails: { ... },
  data: {}
}
```

**Analysis:**

This occurs because `ChatController` actually dispatches the event twice from two different places:

1. As part of the `ChatController.prototype.connect()` call:

https://github.com/amazon-connect/amazon-connect-chatjs/blob/d85c5175d6576a60b6cad2a7b94196116864f4b1/src/core/chatController.js#L192-L201

https://github.com/amazon-connect/amazon-connect-chatjs/blob/d85c5175d6576a60b6cad2a7b94196116864f4b1/src/core/chatController.js#L292-L304

2. Via an event listener of `LpcConnectionHelper.prototype.handleConnectionGain()`:

https://github.com/amazon-connect/amazon-connect-chatjs/blob/d85c5175d6576a60b6cad2a7b94196116864f4b1/src/core/chatController.js#L192-L199

https://github.com/amazon-connect/amazon-connect-chatjs/blob/d85c5175d6576a60b6cad2a7b94196116864f4b1/src/core/chatController.js#L207-L218

https://github.com/amazon-connect/amazon-connect-chatjs/blob/d85c5175d6576a60b6cad2a7b94196116864f4b1/src/core/chatController.js#L247-L252

https://github.com/amazon-connect/amazon-connect-chatjs/blob/d85c5175d6576a60b6cad2a7b94196116864f4b1/src/core/chatController.js#L288-L290

Having duplicate event invocations can be problematic when the user assumes it will be invoked only once and the handler implementation is not idempotent (i.e. invoking it again may have undesired side-effects).

**Proposed fix:**

The event dispatch that is part of `ChatController.prototype.connect()` should be removed, so that only the `LpcConnectionHelper.prototype.handleConnectionGain()` remains. This is because the latter event will be triggered whenever the connection is lost and later regained.

**Comment by sseidel16** [comment]
The code fix does not fix the described issue of CHAT_EVENTS.CONNECTION_ESTABLISHED being fired twice. There is still an event fired from the connect call, and one from LpcConnectionHelper. Perhaps this is desired behavior, but that should be noted. As it stands onConnectionEstablished is still called twice when connect is called a single time.

**Comment by marcogrcr** [comment]
@sseidel16 You are correct. I believe this may have been the result of a mixup. Looks like PR #196 resolved #123 instead of this issue.

**Comment by sseidel16** [comment]
Another interesting note:

The proposed fix in the initial issue is likely not feasible because, when used in conjunction with streams, the websocket can be already connected when connect is called (wsm has already connected). In that scenario, onConnectionEstablished is *only* called by connect, and therefore only invoked once. It seems more reasonable to only emit a connection established event after connect if the websocket has already been started (this can be seen in the response from LpcConnectionHelper start).
